internal/volume/volumes_information.go

```diff
@@ -92,6 +92,9 @@ type BackupVolumeInfo struct {
 	// Snapshot completes timestamp.
 	CompletionTimestamp *metav1.Time `json:"completionTimestamp,omitempty"`
 
+	// Whether the volume data is backed up successfully.
+	Succeeded bool `json:"succeeded,omitempty"`
```

Comment:
Instead of trying to interpret the result we may leave it as is it the specific "XXXInfo" struct.

Reply:
Add the original phase in the `XXXInfo` struct and keep the `Result` parameter.